src/sentry/workflow_engine/endpoints/validators/detector_workflow.py

```diff
@@ -36,6 +41,82 @@ def can_edit_detector(detector: Detector, request: Request) -> bool:
     return False
 
 
+def validate_detectors_exist_and_have_permissions(
```

Comment:
Extracted some common logic out into separate functions so that we could have separate validators for detectors and workflows